pages/blog/integrating-mailchimp-with-your-nodejs-app.mdx

```diff
@@ -0,0 +1,350 @@
+
+- `email_address`: Email address for a subscriber.
+- `status`: Subscriber's current status. Possible values: `subscribed`, `unsubscribed`, `cleaned`, or `pending`. We’ll be using `pending` as the status here since it helps us create a double opt-in list. MailChimp on our behalf will automatically send out the confirmation email if the status is pending. Once a person clicks the link and verifies, their email will show up in your Audience List!
+
+```
+app.post('/subscribe', (req, res) => {
+   const {email} = req.body;
+   const addData = {
+         members: [
+            {
+               email_address: email,
+               status: ‘pending’
+            }
+         ]
+   }
+})
+```
+
+So finally, we need to `JSON.stringify()` this so that it can be passed on to the MailChimp API.
+
+```
+addDataJson = JSON.stringify(addData);
+```
+
+Perfect, so this is done. But this was just the data. Let’s structure the request to be sent as well. We will be making an `options` object for this. This contains the following properties:
+
+- `url`: The API Endpoint we created above containing your `List ID` as well.
+- `method`: This is a post request, so `POST`.
+- `headers`: This contains all the important information like Authorization. We’ll be passing over your `API Key` here.
+- `body`: Here we’ll be passing over the actual body, ie `addDataJson`.
+
+The code will look something like this:
+
+```
+const options = {
+   url: ‘https://us6.api.mailchimp.com/3.0/lists/ac7ad45fa0’,
+   method: ‘POST’,
+   headers: {
+      Authorization: ‘auth <YOUR_API_KEY>’
+   },
+   body: addDataJson
+}
+
+```
+
+Perfect, now it is time for sending this request over to MailChimp, yay! For this we will be using the `request` module we previously imported.
+
+```
+request (options, (error, response, body) => {
+   if(error) {
+      res.json({error}) // error :(
+   } else {
+      res.sendStatus(200); //successful :)
+   }
+})
+```
+
+And that’s it! We’re done with the code here! You have successfully made way for people to opt-in for your mailing list!
+
+## Alternative Method - Using an off the shelf solution
+
+As you can see, just to configure a single API function, we needed to go through a considerable integration pain, which again is a code you need to maintain regularly. What I generally use and recommend to you all is definitely, Grouparoo.
+
+Grouparoo is a data syncing platform that connects directly to your product database, allows you and your team to define what customer data you care about, and then syncs that data to Mailchimp. The best part, the tool is fully open source! You can easily choose between a self-hosted or a cloud offering. Using this will be just a matter of minutes to build this integration, and anyone can easily use the GUI to configure it!
```

Comment:
We have an example app that [connects SQLite data to Mailchimp](https://github.com/grouparoo/app-example-config). Might be useful to link to it here.